crates/bevy_transform/src/components/transform2d.rs

```diff
@@ -0,0 +1,267 @@
+use bevy_ecs::{prelude::Component, reflect::ReflectComponent};
+use bevy_math::{Affine2, Mat2, Mat3, Vec2, Vec3};
+use bevy_reflect::{std_traits::ReflectDefault, FromReflect, Reflect};
+
+/// Describes the position of an [`Entity`] in 2d space.
+///
+/// This component acts as a proxy to the [`Transform`] component,
+/// and thus *requires* that both a [`Transform`] and [`GlobalTransform`] are present to function.
+///
+/// If this [`Transform2d`] has a [`Parent`], then it's relative to the [`Transform2d`] of the [`Parent`].
+#[derive(Component, Debug, PartialEq, Clone, Copy, Reflect, FromReflect)]
+#[reflect(Component, PartialEq, Default)]
+pub struct Transform2d {
+    /// The translation along the `X` and `Y` axes.
+    pub translation: Vec2,
+    /// The rotation in radians. Positive values rotate anti-clockwise.
+    pub rotation: f32,
+    /// The scale along the `X` and `Y` axes.
+    pub scale: Vec2,
+    /// The translation along the `Z` axis.
+    ///
+    /// You might be surprised that 2D entities have a translation along the Z axis,
+    /// but this third dimension is used when rendering to decide what should appear in front or behind.
+    /// A higher translation on the Z axis puts the entity closer to the camera, and thus in front of entities with a lower Z translation.
+    ///
+    /// Keep in mind that this is relative to the [`Parent`]'s `z_translation`.
+    /// The other fields on [`Transform2d`] don't affect this because they are strictly 2D.
+    pub z_translation: f32,
```

Comment:
IMO this is fine as is. james concern is legitimate but it can be relegated to a follow-up PR.

For an entity made out of multiple sprites, you should be able to position on z-axis the sprites relative to each other. It probably needs something like "is this on top or below the parent" rather than a `f32` but I think it's beyond scope.

Reply:
So the idea is that z is used only for sorting, but for rendering, only an Affine2 is needed?